<launch>
    <arg name="sim_time_required" default="true"/>
    <param name="use_sim_time" value="$(arg sim_time_required)"/>

    <arg name="start_rviz" default="true"/>

    <arg name="use_semantics" default="true"/>
    <arg name="semantic_map_path" default="$(find semantic_recolor)/config/colors/ade150_config.csv"/>
    <arg name="pointcloud_topic" default="/semantic_pointcloud"/>
    <arg name="use_gt_frame" default="true"/>
    <arg name="sensor_frame" default="left_cam"/>

    <include file="$(find hydra_utils)/launch/data_sources/d455_pointcloud.launch" pass_all_args="true"/>

    <include file="$(find hydra_topology)/launch/hydra_topology.launch" pass_all_args="true">
        <arg name="config" value="$(find hydra_topology)/config/d455_topology_config.yaml"/>
        <arg name="dsg_output_prefix" value="d455"/>
        <arg name="rviz_file" value="d455.rviz"/>
        <arg name="use_pose_graph" value="$(eval not arg('use_gt_frame'))"/>
    </include>
</launch>
